ocf_datapipes/select/drop_overnight_pv.py

```diff
@@ -1 +1,100 @@
 """Drop PV systems that report overnight"""
+
+import logging
+import random
+from datetime import datetime
+from typing import Dict, List, Union
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+logger = logging.getLogger(__name__)
+
+"""According to the UK Met office- https://www.metoffice.gov.uk/weather/learn-about/met-office-for-schools/other-content/other-resources/our-seasons
+
+Summer  - starts in June in the UK and finishes at the end of August.
+Autumn -  starts in September and finishes in November
+Winter - runs from December to February
+Spring - begins in March and ends in May
+According to worlddata.info - https://www.worlddata.info/europe/united-kingdom/sunset.php
+Month	    Sunrise	    Sunset      Hours of daylight
+January	    07:57 am	04:22 pm	8:25 h
+February	07:12 am	05:16 pm	10:05 h
+March	    06:12 am	06:06 pm	11:54 h
+April	    06:02 am	07:58 pm	13:56 h
+May	        05:06 am	08:47 pm	15:41 h
+June	    04:40 am	09:21 pm	16:41 h
+July	    04:59 am	09:13 pm	16:15 h
+August	    05:44 am	08:25 pm	14:41 h
+September	06:33 am	07:17 pm	12:44 h
+October	    07:22 am	06:09 pm	10:47 h
+November	07:16 am	04:13 pm	8:57 h
+December	07:57 am	03:53 pm	7:56 h
+"""
+
+
+@functional_datapipe("drop_night_pv")
+class DropNightPVIterDataPipe(IterDataPipe):
+    """
+    Drop the pv output of the over night date times in a timeseries Xarray Dataset
+    """
+
+    def __init__(self, source_datapipe: IterDataPipe):
+        """
+        This function provides an extra dimenion of day and night status for each time step
+        in a timeseries Xarray Dataset
+
+        Args
+            source_datapipe: A datapipe that emmits Xarray Dataset of the pv.netcdf file
+        """
+        self.source_datapipe = source_datapipe
+
+    def __iter__(self) -> xr.Dataset():
+        """Function provides ssid's with night time pv output"""
+        logger.warning("This droping of the nighttime pv is only applicable to the UK PV datasets")
+        # Classifying the night time
+        for xr_dataset in self.source_datapipe:
+            xr_dataset = self.source_datapipe
+            dates_list = xr_dataset.coords["datetime"].values.astype("datetime64[s]").tolist()
+
+            def day_status(date_time_of_day: datetime):
+                dtime = date_time_of_day
+                date_month = int(dtime.month)
+                date_hr = int(dtime.strftime("%H"))
+                status_day = "day"
+                if date_month == 1 and (date_hr >= 17 and date_hr <= 7):
```

Comment:
an we put this in a dictionary, so 
 night_time_dict = {'1':[7,17], '2',[7,18],....}
then 
```
for idx in range(0,12):
   if date_month  == night_time_dict[idx]:
       if (date_hr <= night_time_dict[idx][0]) and (date_hr >= night_time_dict[idx][1])
             status_day = "night"
```

It think this would be tidier

Reply:
yes, I think that is much more tidier, as we define a UK seasonal cycle in a dictionary and be done with that, rather than hardcoding! I'll get on with it right away!